edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -424,6 +428,10 @@ export default {
       const preferred = this.$store.getters['plugins/cloudProviderForDriver'](this.provider);
 
       for ( const opt of this.agentArgs['cloud-provider-name'].options ) {
+        if (opt === HARVESTER && this.provider !== opt) {
```

Comment:
cloudProviderForDriver above should return this so that the existing "if" below filters out harvester unless the provider is harvester.

Reply:
@yasker this logic looks like it's needed to ensure the proper filtering is present. Is this straightforward?

Reply:
I will fix it.